src/bika/coa/ajax.py

```diff
@@ -441,6 +444,134 @@ def merge_header_and_values(self,headers,values):
+            if date_verified:
+                all_dates.append(date_verified)
+        all_dates.sort()
+        if len(all_dates) > 1:
+            verified_from = all_dates[0].strftime('%d/%m/20%y')
+            verified_to = all_dates[-1].strftime('%d/%m/20%y')
+        if len(all_dates) == 1:
+            verified_from = all_dates[0].strftime('%d/%m/20%y')
+            verified_to = all_dates[0].strftime('%d/%m/20%y')
+        return verified_from,verified_to
+
+    def get_analyzed_dates(self,samples):
+        analyzed_from = ""
+        analyzed_to = ""
+        all_dates = []
+        for sample in samples:
+            for analysis in sample.Analyses:
+                date_analyzed = analysis.ResultCaptureDate
+                if date_analyzed:
+                    all_dates.append(date_analyzed)
+        all_dates.sort()
+        if len(all_dates) > 1:
+            analyzed_from = all_dates[0].strftime('%d/%m/20%y')
+            analyzed_to = all_dates[-1].strftime('%d/%m/20%y')
+        if len(all_dates) == 1:
+            analyzed_from = all_dates[0].strftime('%d/%m/20%y')
+            analyzed_to = all_dates[0].strftime('%d/%m/20%y')
+        return analyzed_from,analyzed_to
+
+    def get_zlabs_body(self):
+        analysis_services_full_list = api.get_setup().bika_analysisservices.values()
+        eligible_analysis_services = sorted([item for item in analysis_services_full_list if item.getSortKey()], key=lambda x:x.getSortKey())
+        analysis_Ids_list = ["Analysis",""]
+        methods_list = ["Method",""]
+        unit_list = ["Unit",""]
+        final_body_rows = []
+
+        for analysis_service in eligible_analysis_services:
+            if analysis_service.getMethod():
+                methods_list.append(analysis_service.getMethod().Title())
+            else:
+                methods_list.append("")
+            analysis_Ids_list.append(analysis_service.Title())
+            unit_list.append(analysis_service.getUnit())
+        
+        final_body_rows = [
+            analysis_Ids_list,methods_list,unit_list,]
+        return eligible_analysis_services,final_body_rows
+    
+    def get_zlabs_analysis_request(self,samples,analysis_services,extra_column):
+        sorted_samples = sorted(samples, key=lambda x:x.ClientSampleID)
+        sample_data = []
+        sample_analyses,sample_analyses_ids = self.get_sample_analyses(sorted_samples) #The first entry is the sample and the rest are the analyses of those samples
+        for indx,sample in enumerate(sample_analyses):
+            sample_results = [sample[0].ClientSampleID]
+            if extra_column:
+                sample_results.append(sample[0].id)
+            for analysis_service in analysis_services:
+                if analysis_service.getKeyword() in sample_analyses_ids[indx]:
+                    sample_results.append(sample[sample_analyses_ids[indx].index(analysis_service.getKeyword())].getFormattedResult(html=False))
```

Comment:
Maybe instead of one long line you can have
 ```
     keyword = analysis_service.getKeyword()
     if keyword in sample_analyses_ids[indx]:
         analyses_id = sample_analyses_ids[indx]
         sample = sample[analyses_id.index(keyword)]
         sample_result = sample.getFormattedResult(html=False)
         sample_results.append(sample_result)

```